fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/observer/SimpleObserver.java

```diff
@@ -68,6 +81,53 @@
     static final Integer EVENT_TYPES = NODE_ADDED + NODE_REMOVED + NODE_MOVED + PROPERTY_ADDED + PROPERTY_CHANGED
             + PROPERTY_REMOVED;
 
+    private static Stream<FedoraResource> getAllChildren(final FedoraResource resource) {
+        return concat(of(resource), resource.getChildren().flatMap(SimpleObserver::getAllChildren));
+    }
+
+    private static Function<Session, Function<FedoraEvent, FedoraEvent>> namespaceResolver = session -> evt -> {
+        final NamespaceRegistry namespaceRegistry = getNamespaceRegistry(session);
+        final FedoraEvent event = new FedoraEventImpl(evt);
+        evt.getProperties().stream()
+            .filter(property -> validJcrProperties.contains(property) || !property.startsWith("jcr:"))
+            .forEach(property -> {
+                final String[] parts = property.split(":", 2);
+                if (parts.length == 2) {
+                    try {
+                        event.addProperty(
+                            getRDFNamespaceForJcrNamespace(namespaceRegistry.getURI(parts[0])) + parts[1]);
+                    } catch (final RepositoryException ex) {
+                        LOGGER.debug("Prefix could not be dereferenced using the namespace registry: {}", property);
```

Comment:
Is this just a log-and-move-on situation? If the prefix can't be resolved, what exactly are we publishing?